Adding first draft of [OTEL Kubernetes] Cluster Overview Dashboard

gizas opened the pull request:
- Enhancement

## Proposed commit message

Adding a new `[OTEL Kubernetes] Cluster Overview` dashboard in k8s integration

Please explain:

- WHAT: A new dashboard that visualises the needed metrics for K8s Cluster observability
- WHY:  To support the user experience of Kubernetes Observability users that use OTEL collector 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).


## How to test this PR locally

1. Create a kind cluster `kind create cluster`
2. Clone this repo
3. Navigate to `elastic/integrations/packages/kubernetes`
4. Build package by running `elastic-package build`
5. Install the elastic-agent-otel distribution from https://github.com/elastic/k8scollector/blob/otel/helm/elastic-agent/elastic-otel-collector_all.yaml. Run `kubectl apply -f elastic-otel-collector_all.yaml`
6. Spin up a new `elastic-package stack up -d -v --version=8.15.0-SNAPSHOT`
7. The new Kubernetes 1.64.0 integration should be available. Install it
8. Navigate to Dashboards and open `[OTEL Kubernetes] Cluster Overview `


## Screenshots

![Screenshot 2024-07-10 at 1 35 47 PM](https://github.com/elastic/integrations/assets/1685415/5775e8fd-00e8-4551-93e4-e7ee9821ae77)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/13481)
* Commit: 868633db0ba5f0f30ffe6c560205f1a3c5429e62

### Failed CI Steps
* [Check integrations kubernetes](https://buildkite.com/elastic/integrations/builds/13481#01909c46-f56e-49e1-873e-a772095e3f7a)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13481","number":13481,"commit":"868633db0ba5f0f30ffe6c560205f1a3c5429e62"}],"number":13481}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/13606)
* Commit: 3ebdac691db54da3e09c0d3a19c0477f58fc4bb1

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/13606#0190a6c1-3a44-42c1-ad6a-25dd27d158ff)

### History
* :broken_heart: [Build #13481](https://buildkite.com/elastic/integrations/builds/13481) failed 868633db0ba5f0f30ffe6c560205f1a3c5429e62

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13606","number":13606,"commit":"3ebdac691db54da3e09c0d3a19c0477f58fc4bb1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13481","number":13481,"commit":"868633db0ba5f0f30ffe6c560205f1a3c5429e62"}],"number":13606}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/13608)
* Commit: 0cc8911f2aec7def1cf93037123ea90dc65a6920

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/13608#0190a6d2-4f6f-4d73-957c-3ef40fe73a52)

### History
* :broken_heart: [Build #13606](https://buildkite.com/elastic/integrations/builds/13606) failed 3ebdac691db54da3e09c0d3a19c0477f58fc4bb1
* :broken_heart: [Build #13481](https://buildkite.com/elastic/integrations/builds/13481) failed 868633db0ba5f0f30ffe6c560205f1a3c5429e62

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13608","number":13608,"commit":"0cc8911f2aec7def1cf93037123ea90dc65a6920"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13606","number":13606,"commit":"3ebdac691db54da3e09c0d3a19c0477f58fc4bb1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13481","number":13481,"commit":"868633db0ba5f0f30ffe6c560205f1a3c5429e62"}],"number":13608}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/13615)
* Commit: 405ecc02a3bfab4b3401d11cebaf65e50d45d527

### Failed CI Steps
* [:pipeline::arrow_up: Upload Pipeline: .buildkite/pipeline.yml](https://buildkite.com/elastic/integrations/builds/13615#0190a748-6da2-4af8-8569-201a662ceb65)

### History
* :broken_heart: [Build #13608](https://buildkite.com/elastic/integrations/builds/13608) failed 0cc8911f2aec7def1cf93037123ea90dc65a6920
* :broken_heart: [Build #13606](https://buildkite.com/elastic/integrations/builds/13606) failed 3ebdac691db54da3e09c0d3a19c0477f58fc4bb1
* :broken_heart: [Build #13481](https://buildkite.com/elastic/integrations/builds/13481) failed 868633db0ba5f0f30ffe6c560205f1a3c5429e62

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13615","number":13615,"commit":"405ecc02a3bfab4b3401d11cebaf65e50d45d527"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13608","number":13608,"commit":"0cc8911f2aec7def1cf93037123ea90dc65a6920"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13606","number":13606,"commit":"3ebdac691db54da3e09c0d3a19c0477f58fc4bb1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13481","number":13481,"commit":"868633db0ba5f0f30ffe6c560205f1a3c5429e62"}],"number":13615}
buildkite-pr-comment-->

ChrsMark commented on packages/kubernetes/kibana/dashboard/kubernetes-e31db833-de44-4f74-a2df-f7c1337bb1be.json:
```diff
@@ -436,85 +340,1819 @@
+                                                "68bd6f9e-6894-4991-b2ff-fac3a4461b2b": {
+                                                    "dataType": "number",
+                                                    "isBucketed": false,
+                                                    "label": "Average of k8s.node.cpu.utilization",
```
The way it is reported right now by the Collector is not correct. I'd suggest keeping this out for now or use the `k8s.node.cpu.usage` instead.

You can find more info at:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27885#issuecomment-2225039360
- https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6a5b9e673a433ad2c6ed7ef2526a8920fd18cf0d/receiver/kubeletstatsreceiver/metadata.yaml#L96

There is a fix for that at https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/e2483538dde00ad4de2462ff6b22979557880a1f but i'm not sure yet if we will first land this or if we should first deprecate and remove the metrics and then add the correct metric back.
You can follow https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27885 for updates.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/13766)
* Commit: 03e6e8d300aa4dc5f8a50d6c4f32ba1e37ca8416

### History
* :broken_heart: [Build #13615](https://buildkite.com/elastic/integrations/builds/13615) failed 405ecc02a3bfab4b3401d11cebaf65e50d45d527
* :broken_heart: [Build #13608](https://buildkite.com/elastic/integrations/builds/13608) failed 0cc8911f2aec7def1cf93037123ea90dc65a6920
* :broken_heart: [Build #13606](https://buildkite.com/elastic/integrations/builds/13606) failed 3ebdac691db54da3e09c0d3a19c0477f58fc4bb1
* :broken_heart: [Build #13481](https://buildkite.com/elastic/integrations/builds/13481) failed 868633db0ba5f0f30ffe6c560205f1a3c5429e62

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13766","number":13766,"commit":"03e6e8d300aa4dc5f8a50d6c4f32ba1e37ca8416"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13615","number":13615,"commit":"405ecc02a3bfab4b3401d11cebaf65e50d45d527"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13608","number":13608,"commit":"0cc8911f2aec7def1cf93037123ea90dc65a6920"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13606","number":13606,"commit":"3ebdac691db54da3e09c0d3a19c0477f58fc4bb1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13481","number":13481,"commit":"868633db0ba5f0f30ffe6c560205f1a3c5429e62"}],"number":13766}
buildkite-pr-comment-->

gizas commented:
Adding another iteration

- Updated Node visualisation with k8s.node.cpu.usage
- Added tables for memory/ cpu utilisation. Those have coloured cells. Although there seems a problem with colouring. See memory in screenshot, it has red colour although value is 15%
- Added network errors 
- Added node filesystem usage in the bottom

![Screenshot 2024-07-18 at 5 29 51 PM](https://github.com/user-attachments/assets/535615e1-0914-465c-ab27-2cb171bc26d2)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/13778)
* Commit: 9591eee18dfcfad69095b44f77e396b464bbe9e4

### Failed CI Steps
* [Check integrations kubernetes](https://buildkite.com/elastic/integrations/builds/13778#0190c645-fafa-4021-8079-5b764fa3b6c6)

### History
* :green_heart: [Build #13766](https://buildkite.com/elastic/integrations/builds/13766) succeeded 03e6e8d300aa4dc5f8a50d6c4f32ba1e37ca8416
* :broken_heart: [Build #13615](https://buildkite.com/elastic/integrations/builds/13615) failed 405ecc02a3bfab4b3401d11cebaf65e50d45d527
* :broken_heart: [Build #13608](https://buildkite.com/elastic/integrations/builds/13608) failed 0cc8911f2aec7def1cf93037123ea90dc65a6920
* :broken_heart: [Build #13606](https://buildkite.com/elastic/integrations/builds/13606) failed 3ebdac691db54da3e09c0d3a19c0477f58fc4bb1
* :broken_heart: [Build #13481](https://buildkite.com/elastic/integrations/builds/13481) failed 868633db0ba5f0f30ffe6c560205f1a3c5429e62

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13778","number":13778,"commit":"9591eee18dfcfad69095b44f77e396b464bbe9e4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13766","number":13766,"commit":"03e6e8d300aa4dc5f8a50d6c4f32ba1e37ca8416"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13615","number":13615,"commit":"405ecc02a3bfab4b3401d11cebaf65e50d45d527"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13608","number":13608,"commit":"0cc8911f2aec7def1cf93037123ea90dc65a6920"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13606","number":13606,"commit":"3ebdac691db54da3e09c0d3a19c0477f58fc4bb1"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/13481","number":13481,"commit":"868633db0ba5f0f30ffe6c560205f1a3c5429e62"}],"number":13778}
buildkite-pr-comment-->

ChrsMark commented on packages/kubernetes/kibana/dashboard/kubernetes-e31db833-de44-4f74-a2df-f7c1337bb1be.json:
```diff
@@ -0,0 +1,5986 @@
+            {
+                "embeddableConfig": {
+                    "attributes": {
+                        "description": "Average of 100 Top Pod CPU Usage based on kubernetes.pod.usage.limit.pct",
```
If those dashboards aim OTel users the namings should meet the [semantic convention expectations](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#instrument-naming), using the `Utilization` or `Usage` according to the underlying metric's name/notion.

ChrsMark reviewed (commented):
I might miss some content here but something that looks unclear is which type of data this Dashboard aims to consume.

Is this going to use translated metrics coming from the Infra Metrics Processor along with OTel native metrics?

Since this is focused on OTel users and the translation lib emits both the translated and the OTel native ones, why not to use the native ones here directly?

So instead of `kubernetes.pod.cpu.usage.limit.pct` to use the `k8s.pod.cpu_limit_utilization` directly?

gizas commented:
Closed as work was done in different PR
